samples/bluetooth/fast_pair/locator_tag/src/main.c

```diff
@@ -348,24 +353,24 @@ static void fmdn_provisioning_state_changed(bool provisioned)
 
 	/* Triggered on the unprovisioning operation. */
 	if (factory_reset_executed) {
-		LOG_INF("Please press a button to put the device in the Fast Pair discoverable "
-			"advertising mode after a reset to factory settings");
+		LOG_INF("The device was unprovisioned and reset to factory setting");
+
+		/* Disable advertising requests related to the FMDN. */
+		app_fp_adv_request(fp_adv_trigger_ak_write, false);
+		app_fp_adv_request(fp_adv_trigger_clock_sync, false);
+
+		/* Disable advertising request from the UI. */
+		fp_adv_ui_request = false;
+		app_fp_adv_request(fp_adv_trigger_ui, fp_adv_ui_request);
+
 		factory_reset_executed = false;
 
 		return;
 	}
 
-	/* Select the Fast Pair advertising mode according to the FMDN provisioning state. */
-	if (provisioned) {
-		if (is_first_state_changed_cb) {
-			fp_adv_mode = APP_FP_ADV_MODE_NOT_DISCOVERABLE;
-		} else {
-			fp_adv_mode = APP_FP_ADV_MODE_OFF;
-		}
-	} else {
-		fp_adv_mode = APP_FP_ADV_MODE_DISCOVERABLE;
-	}
-	app_fp_adv_mode_set(fp_adv_mode);
+	/* Request the Fast Pair advertising according to the FMDN provisioning state. */
+	app_fp_adv_request(fp_adv_trigger_clock_sync, provisioned && is_first_state_changed_cb);
+	app_fp_adv_request(fp_adv_trigger_ak_write, false);
```

Comment:
Would we advertise right after boot? Maybe we could set the `fmdn_provisioning` flag right on boot when not provisioned?

Reply:
I think it may be a good idea. I think that sample could automatically start advertising right after boot when not provisioned to improve UX.